uswds/uswds
# USWDS - Align alert content with banner

## Summary
Updated the alignment of the alert and site-alert component content to visually match alignment with banner content. 

## Related issue
Closes [#4642](https://github.com/uswds/uswds/issues/4642)

## Preview link
Preview link: [Alert comparison test](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-alert-icon-alignment/?path=/story/components-alert--alert-comparison)

## Problem statement
Design elements like icons and text should align to make it easier for users to scan content. However, the alert, site-alert, and banner components all align along different vertical lines, which can cause visual disruption. 

## Solution
Left-aligning all items that use the `add-responsive-site-margins` mixin along the same vertical line will enable users to more easily scan content. 

### Before
![image](https://user-images.githubusercontent.com/93996430/186020209-39713066-5bd1-4ac2-9865-669f5f383731.png)

### After
![image](https://user-images.githubusercontent.com/93996430/186021550-2b87abd5-52cf-480b-aac9-ac35e93652d8.png)

## Major changes
1. Removed usa-site-alert's dependency on usa-alert. This was done by consolidating all shared styles in uswds-core mixins.
2. Visually aligned the content in both components with the content in usa-banner. 

## Testing and review
To test:
1. Confirm that alert variant styles match existing display. Note: The alignment is expected to be different, but colors, typography, padding, etc should be the same. 
1. Confirm that styles left-align with other components like banner and footer
1. Confirm that alignment looks good in multiple browsers at multiple widths.
1. Confirm that you can disable all styles in usa-alert and still have proper display of usa-site-alert (and vice versa)

---

Before opening this PR, make sure you’ve done whichever of these applies to you:
